build-artifacts/project-template-gradle/build.gradle

```diff
@@ -724,18 +736,20 @@ copyTypings.onlyIf({
 
 task validateAppIdMatch {
```

Comment:
This task appears to be almost identical to setAppIdentifier. You can either extract the common logic in a static function used by both tasks. Or think whether setAppIdentifier should be so early in the configuration phase, because at that point another gradle configuration may overwrite the id.

I'd consider calling setAppIdentifier's code implicitly only on debug builds (meaning put it in a debug-build check) - replacing the app identifier automatically with what's in the package.json, informing the user of the implicit actions, while leaving the application identifier untouched otherwise, if it has already been set someplace else in Gradle.

Reply:
I will extract the 4 common lines of the tasks in a static function.

The behavior you describe is almost the total opposite of what we've decided on the meeting for this feature and what has been approved in the meeting notes from you, @rosen-vladimirov, @PanayotCankov, @Plamen5kov  and other teem members.

We set application identifier that early so that we keep our current behavior and don't force the configuration on the user. If the configuration from package.json is overwritten in later configuration stages the user will get a fair warning that this is not the recommended way and that Livesync will not work properly.

Having the app identifier set one way in debug and another way (and from another place) in release build sounds strange to me. I'd never expect a tooling to behave this way.